File: CPP/141_LinkedListCycle.cpp
// 141 - Linked List Cycle

bool hasCycle(ListNode *head) {
    if(head == nullptr || head->next == nullptr) return false;

    ListNode* a = head, b = head;
    while(a != nullptr && a->next != nullptr) {
        a = a->next->next;
        b = b->next;
        if(a == b) return true;
    }

    return false;
}

File: Java/141_LinkedListCycle.java
// 141 - Linked List Cycle

public boolean hasCycle(ListNode head) {
    if(head == null || head.next == null) return false;

    ListNode a = head, b = head;
    while(a != null && a.next != null) {
        a = a.next.next;
        b = b.next;
        if(a == b) return true;
    }

    return false;
}